### ISSUE TITLE
glibtoolize is broken after reinstalling libtool

### ISSUE THREAD

**Issue by ice0** [Homebrew/homebrew-core#180040]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
brew config
HOMEBREW_VERSION: 4.3.12-102-g78132b1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 78132b13bfa78a2cb481190ac28cdcbc123a31e2
Last commit: 7 hours ago
Core tap HEAD: 62f162b28501fff49b6828dfaa0085014dc9fca2
Core tap last commit: 50 minutes ago
Core tap JSON: 05 Aug 06:06 UTC
Core cask tap HEAD: ee19bb5b6b0ac52aac9b9a8753c909be795a5278
Core cask tap last commit: 3 hours ago
Core cask tap JSON: 05 Aug 06:06 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

```
brew doctor                
Your system is ready to brew.
```

### Verification

- [X] My `brew doctor` output says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.
- [X] I searched for recent similar issues at https://github.com/Homebrew/homebrew-core/issues?q=is%3Aissue and found no duplicates.

### What were you trying to do (and why)?

I am running `glibtoolize --ltdl --force`
What's standard command for autotools build (for copying libtool sources).

### What happened (include all command output)?

```
glibtoolize --ltdl --force 
glibtoolize: putting auxiliary files in 'libltdl'.
glibtoolize: linking file 'libltdl/compile'
glibtoolize: linking file 'libltdl/config.guess'
glibtoolize: linking file 'libltdl/config.sub'
glibtoolize: linking file 'libltdl/depcomp'
glibtoolize: linking file 'libltdl/install-sh'
glibtoolize: linking file 'libltdl/missing'
glibtoolize: linking file 'libltdl/ltmain.sh'
glibtoolize: putting macros in 'libltdl/m4'.
glibtoolize: linking file 'libltdl/m4/libtool.m4'
glibtoolize: linking file 'libltdl/m4/ltargz.m4'
glibtoolize: linking file 'libltdl/m4/ltdl.m4'
glibtoolize: linking file 'libltdl/m4/ltoptions.m4'
glibtoolize: linking file 'libltdl/m4/ltsugar.m4'
glibtoolize: linking file 'libltdl/m4/ltversion.m4'
glibtoolize: linking file 'libltdl/m4/lt~obsolete.m4'
glibtoolize: putting libltdl files in 'libltdl'.
glibtoolize: linking file 'libltdl/COPYING.LIB'
glibtoolize: creating file 'libltdl/Makefile.am'
glibtoolize: linking file 'libltdl/README'
glibtoolize:   error: creating 'libltdl/configure.ac' from '/opt/homebrew/Cellar/libtool/2.4.7/share/libtool/configure.ac' failed
glibtoolize: creating file 'libltdl/aclocal.m4'
glibtoolize: creating file 'libltdl/Makefile.in'
glibtoolize: linking file 'libltdl/config-h.in'
glibtoolize:   error: creating 'libltdl/configure' from '/opt/homebrew/Cellar/libtool/2.4.7/share/libtool/configure' failed
glibtoolize: linking file 'libltdl/libltdl/lt__alloc.h'
glibtoolize: linking file 'libltdl/libltdl/lt__argz_.h'
glibtoolize: linking file 'libltdl/libltdl/lt__dirent.h'
glibtoolize: linking file 'libltdl/libltdl/lt__glibc.h'
glibtoolize: linking file 'libltdl/libltdl/lt__private.h'
glibtoolize: linking file 'libltdl/libltdl/lt__strl.h'
glibtoolize: linking file 'libltdl/libltdl/lt_dlloader.h'
glibtoolize: linking file 'libltdl/libltdl/lt_error.h'
glibtoolize: linking file 'libltdl/libltdl/lt_system.h'
glibtoolize: linking file 'libltdl/libltdl/slist.h'
glibtoolize: linking file 'libltdl/loaders/dld_link.c'
glibtoolize: linking file 'libltdl/loaders/dlopen.c'
glibtoolize: linking file 'libltdl/loaders/dyld.c'
glibtoolize: linking file 'libltdl/loaders/load_add_on.c'
glibtoolize: linking file 'libltdl/loaders/loadlibrary.c'
glibtoolize: linking file 'libltdl/loaders/preopen.c'
glibtoolize: linking file 'libltdl/loaders/shl_load.c'
glibtoolize: linking file 'libltdl/lt__alloc.c'
glibtoolize: linking file 'libltdl/lt__argz.c'
glibtoolize: linking file 'libltdl/lt__dirent.c'
glibtoolize: linking file 'libltdl/lt__strl.c'
glibtoolize: linking file 'libltdl/lt_dlloader.c'
glibtoolize: linking file 'libltdl/lt_error.c'
glibtoolize: linking file 'libltdl/ltdl.c'
glibtoolize: linking file 'libltdl/ltdl.h'
glibtoolize: linking file 'libltdl/slist.c'
```

### What did you expect to happen?

glibtoolize should not fail

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew reinstall libtool
glibtoolize --ltdl --force
```

**Comment by ice0** [comment]
I did some checks before submitting the issue.
glibtoolize fails due to `sed` command.

```
glibtoolize --ltdl --force -n -v | grep "configure.ac" -B2
...
rm -f libltdl/configure.ac
glibtoolize: /usr/bin/sed -e '
glibtoolize:           s||.|g' /opt/homebrew/Cellar/libtool/2.4.7/share/libtool/configure.ac > libltdl/configure.ac
```

If you try to run this command to reproduce the behavior of `glibtoolize`, it will fail:
```
/usr/bin/sed -e 's||.|g' /opt/homebrew/Cellar/libtool/2.4.7/share/libtool/configure.ac > libltdl/configure.ac
sed: first RE may not be empty
```

This is how `glibtoolize` output looks like:
```
bash -x glibtoolize --verbose --ltdl --force -n &>./out.txt
grep "s||.|g" out.txt -B2

...
func_notquiet '/usr/bin/sed -e '\''
          s||.|g'\'' /opt/homebrew/Cellar/libtool/2.4.7/share/libtool/configure > libltdl/configure'
```

This is how working `glibtoolize` should look like:
```
func_notquiet '/usr/bin/sed -e '\''
          s|\.\.\/build-aux|.|g'\'' /opt/homebrew/Cellar/libtool/2.4.7/share/libtool/configure > libltdl/configure'
```

For some reason the paths are empty.

**Comment by carlocab** [comment]
This seems to be due to the rebuild at #176589. Not sure why, though, since nothing relevant was changed: 01839ee9ff819b9886f7c5e036844e11153d0e03

@gromgit, mind taking a look at this?

**Comment by carlocab** [comment]
Here is a hunk from the diff between a good (old) and bad (new) build of `libtool`:
```diff
@@ -4365,15 +4365,15 @@
 # installed libltdl/configure.ac, then we already know what value
 # we used.
 require_ltdl_ac_aux_dir=func_require_ltdl_ac_aux_dir
 func_require_ltdl_ac_aux_dir ()
 {
     $debug_cmd

-    ltdl_ac_aux_dir="../build-aux"
+    ltdl_ac_aux_dir=""
     func_verbose "ltdl_ac_aux_dir='$ltdl_ac_aux_dir'"

     require_ltdl_ac_aux_dir=:
 }


 # require_ltdl_dir
```
Not sure why `ltdl_ac_aux_dir` is now blank.

**Comment by gromgit** [comment]
@carlocab, where did you get the good build from? I've confirmed that the problem is isolated to macOS builds, as the Linux `libtoolize` does indeed set `lidl_ac_aux_dir="../build-aux"` as above.

**Comment by cho-m** [comment]
This is most likely the `m4` shim causing a mix up of Xcode and brew `m4`. Sadly Xcode CLT is missing `m4` so we can't drop the shim. Someone else may want to see if there is a better way to handle logic to avoid this.

Easiest fix for now is bypassing shim via `M4` - #180067
